src/test/java/com/pubnub/contract/objectV2/step/ThenSteps.kt

```diff
@@ -0,0 +1,81 @@
+
+        val actualPNUUIDMetadata = setUUIDMetadataState.result!!.data
+
+        assertEquals(expectedPNUUIDMetadata.id, actualPNUUIDMetadata.id)
+        assertEquals(expectedPNUUIDMetadata.name, actualPNUUIDMetadata.name)
+        assertEquals(expectedPNUUIDMetadata.email, actualPNUUIDMetadata.email)
+        assertEquals(expectedPNUUIDMetadata.externalId, actualPNUUIDMetadata.externalId)
+        assertEquals(expectedPNUUIDMetadata.profileUrl, actualPNUUIDMetadata.profileUrl)
+        assertEquals(expectedPNUUIDMetadata.custom, actualPNUUIDMetadata.custom)
+        assertEquals(expectedPNUUIDMetadata.updated, actualPNUUIDMetadata.updated)
+        assertEquals(expectedPNUUIDMetadata.eTag, actualPNUUIDMetadata.eTag)
+    }
+
+    @Then("the UUID metadata for {string} and {string} persona")
+    fun the_UUID_metadata_for_first_and_second_persona(persona01Name: String, persona02Name: String) {
+        val expectedPNUUIDMetadataForAlice: PNUUIDMetadata = loadPersona(persona01Name)
+        val expectedPNUUIDMetadataForJames: PNUUIDMetadata = loadPersona(persona02Name)
+
+        val actualPNUUIDMetadataForFirstPersona = getAllUUIDMetadataState.result!!.data[0]
+        val actualPNUUIDMetadataForSecondPersona = getAllUUIDMetadataState.result!!.data[1]
+
+        assertEquals(expectedPNUUIDMetadataForAlice.id, actualPNUUIDMetadataForFirstPersona.id)
+        assertEquals(expectedPNUUIDMetadataForAlice.name, actualPNUUIDMetadataForFirstPersona.name)
+        assertEquals(expectedPNUUIDMetadataForAlice.email, actualPNUUIDMetadataForFirstPersona.email)
+        assertEquals(expectedPNUUIDMetadataForAlice.externalId, actualPNUUIDMetadataForFirstPersona.externalId)
+        assertEquals(expectedPNUUIDMetadataForAlice.profileUrl, actualPNUUIDMetadataForFirstPersona.profileUrl)
+        assertEquals(expectedPNUUIDMetadataForAlice.custom, actualPNUUIDMetadataForFirstPersona.custom)
+        assertEquals(expectedPNUUIDMetadataForAlice.updated, actualPNUUIDMetadataForFirstPersona.updated)
+        assertEquals(expectedPNUUIDMetadataForAlice.eTag, actualPNUUIDMetadataForFirstPersona.eTag)
+
+        assertEquals(expectedPNUUIDMetadataForJames.id, actualPNUUIDMetadataForSecondPersona.id)
+        assertEquals(expectedPNUUIDMetadataForJames.name, actualPNUUIDMetadataForSecondPersona.name)
+        assertEquals(expectedPNUUIDMetadataForJames.email, actualPNUUIDMetadataForSecondPersona.email)
+        assertEquals(expectedPNUUIDMetadataForJames.externalId, actualPNUUIDMetadataForSecondPersona.externalId)
+        assertEquals(expectedPNUUIDMetadataForJames.profileUrl, actualPNUUIDMetadataForSecondPersona.profileUrl)
+        assertEquals(expectedPNUUIDMetadataForJames.custom, actualPNUUIDMetadataForSecondPersona.custom)
+        assertEquals(expectedPNUUIDMetadataForJames.updated, actualPNUUIDMetadataForSecondPersona.updated)
+        assertEquals(expectedPNUUIDMetadataForJames.eTag, actualPNUUIDMetadataForSecondPersona.eTag)
```

Comment:
That can be significantly shorter and IMHO better from the point of view of information you're getting when it'll fail

```
assertThat(getAllUUIDMetadataState.result!!.data, Matchers.containsInAnyOrder(firstPersona, secondPersona))
```

Question is should we care about the order? IMHO it's not important here, but if you think differently then we can talk about it (maybe including others)

Reply:
I don't think order is important here. I like proposed approach.